app/scripts/controllers/transactions/index.js

```diff
@@ -146,6 +150,49 @@ export default class TransactionController extends EventEmitter {
     return integerChainId;
   }
 
+  /**
+   * @ethereumjs/tx uses @ethereumjs/common as a configuration tool for
+   * specifying which chain, network, hardfork and EIPs to support for
+   * a transaction. By referencing this configuration, and analyzing the fields
+   * specified in txParams, @ethereumjs/tx is able to determine which EIP-2718
+   * transaction type to use.
+   * @returns {Common} common configuration object
+   */
+  getCommonConfiguration() {
+    // type will be one of our default network names or 'rpc'. the default
+    // network names are sufficient configuration, simply pass the name as the
+    // chain argument in the constructor.
+    const { type, nickname } = this.getProviderConfig();
+
+    if (type === 'rpc') {
+      // For 'rpc' we need to use the same basic configuration as mainnet,
+      // since we only support EVM compatible chains, and then override the
+      // name, chainId and networkId properties. This is done using the
+      // `forCustomChain` static method on the Common class.
+      const chainId = this._getCurrentChainId();
+      const networkId = this.networkStore.getState();
+
+      const customChainParams = {
+        name: nickname,
+        chainId: parseInt(chainId, 16),
+        // It is improbable for a transaction to be signed while the network
+        // is loading for two reasons.
+        // 1. Pending, unconfirmed transactions are wiped on network change
+        // 2. The UI is unusable (loading indicator) when network is loading.
+        // seeing the networkId to 0 is for type safety and to explicity lead
+        // the transaction to failing if a user is able to get to this branch
+        // on a custom network that requires valid network id. I have not ran
+        // into this limitation on any network I have attempted, even when
+        // hardcoding networkId to 'loading'.
+        networkId: networkId === 'loading' ? 0 : parseInt(networkId, 10),
+      };
+
+      return Common.forCustomChain('mainnet', customChainParams, HARDFORK);
+    }
+
+    return new Common({ chain: type, hardfork: HARDFORK });
```

Comment:
Also minor, (and our linter may not like this, I forget) Can we check for `type !== 'rpc'` at the top of the function and `return new Common({ chain: type, hardfork: HARDFORK });` immediately, then we don't to have the rpc type logic in a conditional? Totally minor and probably too picky :)

Reply:
Done! Good call. I also changed 'rpc' -> NETWORK_TYPE_RPC